base/sparse/sparsematrix.jl

```diff
@@ -42,7 +42,7 @@ julia> nnz(A)
 3
 ```
 """
-nnz(S::SparseMatrixCSC) = Int(S.colptr[end]-1)
+nnz(S::SparseMatrixCSC) = Int(S.colptr[S.n + 1]-1)
```

Comment:
the distinction here could use a more direct test that didn't go through cholmod-specific methods